sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TimestampFormatter.scala

```diff
@@ -61,8 +64,23 @@ class Iso8601TimestampFormatter(
     needVarLengthSecondFraction: Boolean)
   extends TimestampFormatter with DateTimeFormatterHelper {
   @transient
-  protected lazy val formatter: DateTimeFormatter =
-    getOrCreateFormatter(pattern, locale, needVarLengthSecondFraction)
+  protected lazy val formatter: DateTimeFormatter = {
+    try {
```

Comment:
can we follow `checkDiffResult` and create a common method for both `TimestampFormatter` and `DateFormatter`?

Reply:
OK